### PR TITLE
Fixed issue: throw IOException: The process cannot access the file 'xxxx\log\Non-Session-Log.messages.current.log' because it is being used by another process.

### PR DESCRIPTION
**Pull request by VAllens**
- throw System.IOException: The process cannot access the file 'xxxx\log\Non-Session-Log.messages.current.log' because it is being used by another process. (**Validation using [ThreadedSocketAcceptorTests.TestRecreation](https://github.com/connamara/quickfixn/blob/2d544b5ffd89e7ac57e454aa985b17eae72460a5/UnitTests/ThreadedSocketAcceptorTests.cs#L49C21-L49C35) unit test method**)
- When the `Acceptor` stops, the `_isStarted` field value should be set to `false`.
- Since .NET runtime’s exceptions use culture information to set the Message value, the message value of the Exception should not be asserted, but rather the ErrorCode of the Exception. (**Verify by switching the regional settings of the operating system to a non-English speaking country.**)

### CODE REVIEW

File: QuickFIXn/ThreadedSocketAcceptor.cs [hunk 1]
```diff
@@ -274,6 +274,8 @@ public void Stop(bool force)
             LogoutAllSessions(force);
             DisposeSessions();
             _sessions.Clear();
+            _nonSessionLog.Dispose();
+            _isStarted = false;
```
Comment (VAllens) [inline QuickFIXn/ThreadedSocketAcceptor.cs:278]: When the `Acceptor` stops, the `_isStarted` field value should be set to `false`

File: QuickFIXn/ThreadedSocketAcceptor.cs [hunk 2]
```diff
@@ -274,6 +274,8 @@ public void Stop(bool force)
             LogoutAllSessions(force);
             DisposeSessions();
             _sessions.Clear();
+            _nonSessionLog.Dispose();
```
Comment (VAllens) [inline QuickFIXn/ThreadedSocketAcceptor.cs:277]: The `_nonSessionLog` field should be disposed when the `Acceptor` stops.

**Comment by VAllens** [comment]
This pull request is to fix this issue: https://github.com/connamara/quickfixn/issues/891

**Comment by gbirchmeier** [comment]
On Windows I am able to replicate your the #891 bug.  (Kind of embarrassed about that one)

However, when I run tests on this PR branch I'm still seeing issues.  The tests themselves pass, but there's a failure on cleanup.  It's not the same every time, but it's always variations of ObjectDisposedException.  I'm still investigating.

**Comment by gbirchmeier** [comment]
The exceptions I'm seeing are triggered when I run a sequence of SessionDynamicTest tests.  Running just "DifferentPortForAcceptorTest" followed by "DynamicInitiator" will trigger it.

**Comment by gbirchmeier** [comment]
You're right, all tests pass in VS.

In addition to failing `dotnet test`, it also fails in Jetbrains Rider (my preferred IDE) on Windows.  (Probably Rider is using `dotnet test` under the hood.)

At any rate, I need to fix that.  The UTs need to pass on all these platforms.

**Comment by VAllens** [comment]
> You're right, all tests pass in VS.
> 
> 
> 
> In addition to failing `dotnet test`, it also fails in Jetbrains Rider (my preferred IDE) on Windows.  (Probably Rider is using `dotnet test` under the hood.)
> 
> 
> 
> At any rate, I need to fix that.  The UTs need to pass on all these platforms.
> 
> 

Okay

**Comment by VAllens** [comment]
> You're right, all tests pass in VS.
> 
> In addition to failing `dotnet test`, it also fails in Jetbrains Rider (my preferred IDE) on Windows. (Probably Rider is using `dotnet test` under the hood.)
> 
> At any rate, I need to fix that. The UTs need to pass on all these platforms.

Hi, Bro. :)
Any luck?

**Comment by VAllens** [comment]
Initial analysis shows that the `ObjectDisposedException` is thrown when the [SessionDynamicTest.DynamicInitiator](https://github.com/connamara/quickfixn/blob/c055302218dff16405eab294c64afca6957cc1cb/UnitTests/SessionDynamicTest.cs#L471) method is executed.

Upon further analysis, one of the following two modifications will allow the dotnet test command to work properly and no longer throw exceptions:

1. Comment line [108](https://github.com/connamara/quickfixn/blob/c055302218dff16405eab294c64afca6957cc1cb/QuickFIXn/SocketInitiatorThread.cs#L108) of the `SocketInitiatorThread.cs` file.

2. Add OperationCanceledException exception recognition to line [155](https://github.com/connamara/quickfixn/blob/c055302218dff16405eab294c64afca6957cc1cb/QuickFIXn/SocketInitiatorThread.cs#L155) of the `SocketInitiatorThread.cs` file.
```csharp
_currentReadTask = null;

if (ex.InnerException is OperationCanceledException)
{
    // Nothing read
    return 0;
}

var ioException = ex.InnerException as IOException;
```

Also, I found out that an exception actually occurs when executing the test on the Visual Studio UI, except that the exception is thrown at the end of the test, so it's not caught by the test framework.

**Pull request closed without merging**